tests/testAutopas/tests/containers/TraversalComparison.cpp

```diff
@@ -0,0 +1,236 @@
+  for (auto &elem : shiftVectorByID) {
+    elem = randomShift(magnitude, generator);
+  }
+  unsigned long numIteratedParticles = 0;
+  for (auto &mol : *containerPtr) {
+    mol.addR(shiftVectorByID[mol.getID()]);
+    ++numIteratedParticles;
+  }
+  EXPECT_EQ(numIteratedParticles, totalNumParticles);
+}
+
+/**
+ * Calculates the forces for a given configuration.
+ * @param containerOption Specifies the container.
+ * @param traversalOption Specifies the traversal.
+ * @param dataLayoutOption Specifies the data layout.
+ * @param newton3Option Specifies whether the newton3 optimization should be used or not.
+ * @param numMolecules The number of molecules.
+ * @param numHaloMolecules The number of halo molecules.
+ * @param boxMax The maximum of the simulation box. The minimum is {0.,0.,0.}
+ * @param cellSizeFactor The cell size factor.
+ * @param doSlightShift Specifies whether to add random shifts of size skin/2 to all particles after the neighbor list
+ * generation.
+ * @return Tuple of forces for all particles, ordered by particle id, and array of the globals values (upot, double).
+ */
+std::tuple<std::vector<std::array<double, 3>>, std::array<double, 2>> TraversalComparison::calculateForces(
+    autopas::ContainerOption containerOption, autopas::TraversalOption traversalOption,
+    autopas::DataLayoutOption dataLayoutOption, autopas::Newton3Option newton3Option, unsigned long numMolecules,
+    unsigned long numHaloMolecules, std::array<double, 3> boxMax, double cellSizeFactor, bool doSlightShift) {
+  // Construct container
+  autopas::ContainerSelector<Molecule, FMCell> selector{_boxMin, boxMax, _cutoff};
+  double skin = _cutoff * 0.1;
+  selector.selectContainer(containerOption, autopas::ContainerSelectorInfo{cellSizeFactor, skin, 32});
+  auto container = selector.getCurrentContainer();
+  autopas::LJFunctor<Molecule, FMCell, true /*applyShift*/, false /*useMixing*/, autopas::FunctorN3Modes::Both,
+                     true /*calculateGlobals*/>
+      functor{_cutoff};
+  functor.setParticleProperties(_eps * 24, _sig * _sig);
+
+  auto traversal = autopas::TraversalSelector<FMCell>::generateTraversal(
+      traversalOption, functor, container->getTraversalSelectorInfo(), dataLayoutOption, newton3Option);
+  if (not traversal->isApplicable()) {
+    return {};
+  }
+
+  autopasTools::generators::RandomGenerator::fillWithParticles(*container, Molecule({0., 0., 0.}, {0., 0., 0.}, 0),
+                                                               container->getBoxMin(), container->getBoxMax(),
+                                                               numMolecules);
+
+  autopasTools::generators::RandomGenerator::fillWithHaloParticles(
+      *container, Molecule({0., 0., 0.}, {0., 0., 0.}, numMolecules), container->getCutoff(), numHaloMolecules);
+
+  container->rebuildNeighborLists(traversal.get());
+
+  if (doSlightShift) {
+    executeSlightShift(container, skin / 2, numMolecules + numHaloMolecules);
```

Comment:
`skin / 2` could also be replaced by a `constexpr` but ok...

Reply:
ignored